src/main/java/org/jenkinsci/plugins/pagerduty/changeevents/ChangeEventBuilder.java

```diff
@@ -73,39 +97,71 @@ public void setSummaryText(String summaryText) {
 	this.summaryText = summaryText;
     }
 
-    @Extension
-    public static final class DescriptorImpl extends BuildStepDescriptor<Builder> {
-	@Nonnull
-	public String getDisplayName() {
-	    return "PagerDuty Change Event";
+	@DataBoundSetter
+	public void setCustomDetails(String customDetails) {
+		this.customDetails = customDetails;
 	}
 
-	@Override
-	public boolean isApplicable(Class<? extends AbstractProject> aClass) {
-	    return true;
+	public String getCustomDetails() {
+		return customDetails;
 	}
 
-	/**
-	 * Provides basic validation of integration keys.
-	 *
-	 * Just ensures they are the correct length and only include allowed characters.
-	 *
-	 * @param value The integration key
-	 * @return Whether or not the integration key is valid
-	 */
-	public FormValidation doCheckIntegrationKey(@QueryParameter String value) {
-	    Pattern pattern = Pattern.compile("^[0-9a-z]{32}$");
-	    Matcher matcher = pattern.matcher(value);
+    @Extension
+    public static final class DescriptorImpl extends BuildStepDescriptor<Builder> {
+		@Nonnull
+		public String getDisplayName() {
+			return "PagerDuty Change Event";
+		}
 
-	    if (matcher.matches()) {
-		return FormValidation.ok();
-	    }
+		@Override
+		public boolean isApplicable(Class<? extends AbstractProject> aClass) {
+			return true;
+		}
 
-	    if (value.length() != 32) {
-		return FormValidation.error("Must be 32 characters long");
-	    }
+		/**
+		 * Provides basic validation of integration keys.
+		 *
+		 * Just ensures they are the correct length and only include allowed characters.
+		 *
+		 * @param value The integration key
+		 * @return Whether or not the integration key is valid
+		 */
+		public FormValidation doCheckIntegrationKey(@QueryParameter String value) {
+			if (value.startsWith("$")) {
+				// token expansion so don't validate on length
+				return FormValidation.ok();
+			}
+
+			Pattern pattern = Pattern.compile("^[0-9a-z]{32}$");
+			Matcher matcher = pattern.matcher(value);
+
+			if (matcher.matches()) {
+			return FormValidation.ok();
+			}
+
+			if (value.length() != 32) {
+			return FormValidation.error("Must be 32 characters long");
+			}
+
+			return FormValidation.error("Must only be letters and digits");
+		}
 
-	    return FormValidation.error("Must only be letters and digits");
-	}
+		/**
+		 * Provides basic validation of custom details to ensure they are valid JSON
+		 *
+		 * @param value The custom details
+		 * @return Whether or not the custom details are valid JSON
+		 */
+		public FormValidation doCheckCustomDetails(@QueryParameter String value) {
+			try {
+				objectMapper.reader(DeserializationFeature.FAIL_ON_TRAILING_TOKENS, DeserializationFeature.FAIL_ON_READING_DUP_TREE_KEY).readTree(value);
```

Comment:
Just want to verify input is valid JSON.  The DeserializationFeature settings are extensive so I went with these but am open to suggestions